sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -95,41 +95,6 @@ def test_imputation_shape():
         assert_equal(X_imputed.shape, (10, 2))
 
 
-@ignore_warnings
-def test_imputation_mean_median_only_zero():
```

Comment:
It might be best to not change the deprecated Imputer or its behaviours.

For testing SimpleImputer shouldn't we still test it on the columns that do not contain NaN

Reply:
One could argue that it's a bug as it is now. If we say that "missing values are represented by `0`", then what are `np.nan`? check_array will just not accept nans then. This is why we agreed to use `allow-nan` only if the missing values are `np.nan` right?

It certainly won't work in MICE to have `np.nan` in this way, and I think the two imputers should have consistent behavior, even if it's the deprecated one.